src/common/variables.c

```diff
@@ -252,6 +255,40 @@ static char *get_base_value(dt_variables_params_t *params, char **variable)
     result = g_strdup(params->data->camera_alias);
   else if(has_prefix(variable, "ID"))
     result = g_strdup_printf("%d", params->imgid);
+  else if(has_prefix(variable, "VERSION_NAME"))
+  {
+    GList *res = dt_metadata_get(params->imgid, "Xmp.darktable.version_name", NULL);
+    res = g_list_first(res);
+    if(res != NULL)
+    {
+      result = g_strdup((char *)res->data);
+    }
+    g_list_free_full(res, &g_free);
+  }
+  else if(has_prefix(variable, "VERSION_IF_MULTI"))
+  {
+    sqlite3_stmt *stmt;
+
+    // count duplicates
+    DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
```

Comment:
is not simpler something like that:
SELECT COUNT(*) FROM main.images AS i1
JOIN main.images AS i2 ON i2.film_id = i1.film_id AND i2.filename = i1.filename 
WHERE i1.id = ?1

?

Reply:
Gives the same result. Don't know enough about sqlite to know if it's more efficient (I'm an Oracle developer in my day job). Happy to change if you think it's better.

Reply:
I don't know neither. The syntax just more familiar to me.